app/src/main/java/com/example/sarahtang/mdbsocials/DetailsActivity.java

```diff
@@ -0,0 +1,220 @@
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.support.annotation.NonNull;
+import android.support.design.widget.FloatingActionButton;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.bumptech.glide.Glide;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.MutableData;
+import com.google.firebase.database.Transaction;
+import com.google.firebase.storage.FirebaseStorage;
+
+import java.util.ArrayList;
+
+/**
+ * Details Activity: Detailed information about each social
+ */
+
+public class DetailsActivity extends AppCompatActivity implements View.OnClickListener {
+
+    ImageView imageDetail;
+    TextView nameDetail; //title
+    TextView interestedDetail;//number of people interested
+    TextView hostDetail;
+    TextView descriptionDetail;
+    String firebaseKey;
+    ArrayList<String> peopleInterested;
+    String currEmail = FirebaseAuth.getInstance().getCurrentUser().getEmail();
+    boolean buttonPressed = false;
+    boolean numberChanged = false;
+    FloatingActionButton fabInterested;
+    FloatingActionButton fabStar;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_details);
+
+        /**
+         * On Click Listener implementation in activity
+         */
+
+
+        fabInterested = (FloatingActionButton) findViewById(R.id.fabInterested);
+        fabInterested.setOnClickListener(this);
+        fabStar = (FloatingActionButton) findViewById(R.id.fabStar);
+        fabStar.setOnClickListener(this);
+
+        imageDetail = (ImageView) findViewById(R.id.imageDetail);
+        nameDetail = (TextView) findViewById(R.id.nameDetail);
+        interestedDetail = (TextView) findViewById(R.id.interestedDetail);
+        hostDetail = (TextView) findViewById(R.id.hostDetail);
+        descriptionDetail = (TextView) findViewById(R.id.descriptionDetail);
+
+        //Getting information from intent from socials adapter
+        Intent intent = getIntent();
+        String name = intent.getStringExtra(SocialsAdapter.NAME_KEY);
+        String host = intent.getStringExtra(SocialsAdapter.HOST_KEY);
+        String description = intent.getStringExtra(SocialsAdapter.DESCRIPTION_KEY);
+        String image = intent.getStringExtra(SocialsAdapter.FIREBASE_URL);
+        final int numberInterested = intent.getIntExtra(SocialsAdapter.NUMBER_INTERESTED, -1);
```

Comment:
What is -1, I can't tell (is it the default value?). Specify in a final variable what this is. Maybe call it DEFAULT_INTERESTED so it's easier to read and easier to change later.